test/unit/org/apache/cassandra/dht/BootStrapperTest.java

```diff
@@ -83,6 +104,44 @@ public void testSourceTargetComputation() throws UnknownHostException
         }
     }
 
+    @Test
+    @BMRule(name = "Make sure the non-optimized path is picked up for some operations",
+    targetClass = "org.apache.cassandra.dht.RangeStreamer",
+    targetMethod = "convertPreferredEndpointsToWorkMap(EndpointsByReplica)",
+    action = "org.apache.cassandra.dht.BootStrapperTest.optimizationSkipped()")
+    public void testOptmizationsSkip() throws UnknownHostException
+    {
+        testSkipFFOptmizationFeatureFlag(true);
+    }
+
+    private void testSkipFFOptmizationFeatureFlag(boolean skip) throws UnknownHostException
+    {
+        final int[] clusterSizes = new int[]{ 1, 3, 5, 10, 100 };
```

Comment:
:question
Given the test doesn't assert on the actual computation results but on whether a method call was invoked, is there value in running it for all of the values? I understand it has to be > 1, wouldn't 100 be sufficient?

Reply:
Actually that's a left-over and we only test once